gneiss/regression/tests/test_regression.py

```diff
@@ -0,0 +1,118 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import unittest
+import subprocess
+
+import pandas as pd
+import pandas.util.testing as pdt
+from skbio.util import get_data_path
+from gneiss.regression._regression import lme_regression, ols_regression
+from gneiss.regression.tests.test_ols import TestOLS
+from gneiss.regression.tests.test_mixedlm import TestMixedLM
+from qiime2.metadata import Metadata
+
+
+class TestOLSPlugin(TestOLS):
+
+    def test_ols_regression(self):
+        m = Metadata(self.metadata)
+        res = ols_regression(self.table, self.tree, m, 'real')
+        res_coef = res.coefficients()
+        exp_coef = pd.DataFrame(
+            {'Intercept': [0, 1.00],
+             'real': [1.0, 0]},
+            index=['Y1', 'Y2'])
+
+        pdt.assert_frame_equal(res_coef, exp_coef,
+                               check_exact=False,
+                               check_less_precise=True)
+        # Double check to make sure the fit is perfect
+        self.assertAlmostEqual(res.r2, 1)
+
+        # Double check to make sure residuals are zero
+        exp_resid = pd.DataFrame([[0., 0.],
+                                  [0., 0.],
+                                  [0., 0.],
+                                  [0., 0.],
+                                  [0., 0.]],
+                                 index=['s1', 's2', 's3', 's4', 's5'],
+                                 columns=['Y1', 'Y2'])
+        pdt.assert_frame_equal(exp_resid, res.residuals())
+
+    def test_ols_cli(self):
+        # TODO: Is there a better way to test this?
+        in_table = get_data_path("test_ols_composition.qza")
+        in_tree = get_data_path("test_tree.qza")
+        in_metadata = get_data_path("test_ols_metadata.txt")
+
+        cmd = ("qiime gneiss ols-regression "
+               "--i-table %s "
+               "--i-tree %s "
+               "--o-linear-model test_ols "
+               "--p-formula 'ph' "
+               "--m-metadata-file %s")
+        proc = subprocess.Popen(cmd % (in_table, in_tree, in_metadata),
+                                shell=True)
+        proc.wait()
+
+        self.assertTrue(os.path.exists("test_ols.qza"))
+
```

Comment:
can you also add a test to compare the file contents? Maybe a call to a unix diff?